pkg/ottl/functions.go

```diff
@@ -261,3 +285,15 @@ func buildSlice[T any](argVal value, argType reflect.Type, buildArg buildArgFunc
 
 	return vals, nil
 }
+
+// CreateFunctionMap takes a list of factories and returns a map of Factories
+// keyed on their canonical names.
+func CreateFunctionMap[K any](functions []Factory[K]) map[string]Factory[K] {
```

Comment:
If we make this variadic I think it cleans up its use as we won't have to define the list each time we call it.

Reply:
Great catch, not sure why that didn't occur to me. Done.